src/main/java/org/jabref/gui/externalfiles/AutoLinkFilesAction.java

```diff
@@ -34,47 +33,62 @@
     private final UndoManager undoManager;
     private final TaskExecutor taskExecutor;
 
-    public AutoLinkFilesAction(JabRefFrame frame, PreferencesService preferences, StateManager stateManager, UndoManager undoManager, TaskExecutor taskExecutor) {
-        this.dialogService = frame.getDialogService();
+    public AutoLinkFilesAction(DialogService dialogService, PreferencesService preferences, StateManager stateManager, UndoManager undoManager, TaskExecutor taskExecutor) {
+        this.dialogService = dialogService;
         this.preferences = preferences;
         this.stateManager = stateManager;
         this.undoManager = undoManager;
         this.taskExecutor = taskExecutor;
 
         this.executable.bind(needsDatabase(this.stateManager).and(needsEntriesSelected(stateManager)));
         this.statusMessage.bind(BindingsHelper.ifThenElse(executable, "", Localization.lang("This operation requires one or more entries to be selected.")));
-
     }
 
     @Override
     public void execute() {
-        BibDatabaseContext database = stateManager.getActiveDatabase().orElseThrow(() -> new NullPointerException("Database null"));
-        List<BibEntry> entries = stateManager.getSelectedEntries();
-
-        final NamedCompound nc = new NamedCompound(Localization.lang("Automatically set file links"));
-        AutoSetFileLinksUtil util = new AutoSetFileLinksUtil(
+        final BibDatabaseContext database = stateManager.getActiveDatabase().orElseThrow(() -> new NullPointerException("Database null"));
+        final List<BibEntry> entries = stateManager.getSelectedEntries();
+        final AutoSetFileLinksUtil util = new AutoSetFileLinksUtil(
                 database,
                 preferences.getFilePreferences(),
                 preferences.getAutoLinkPreferences(),
                 ExternalFileTypes.getInstance());
-        Task<List<BibEntry>> linkFilesTask = new Task<>() {
+        final NamedCompound nc = new NamedCompound(Localization.lang("Automatically set file links"));
+
+        if (entries.stream().mapToInt(entry -> entry.getFiles().size()).sum() == 0) {
```

Comment:
`entries.stream().anyMatch(entry -> !entries.getFiles().isEmpty())` is probably easier to understand.

Reply:
But why is this check added in the first place? The autolinkfiles feature tries to find and link new files, so I don't understand why the entry already needs to have a file linked?

Reply:
As I understand this feature, this calls AutoLinkFilesLinkUtil::findAssociatedNotLinkedFiles to check for broken file links. But those are also represented by a LinkedFile. So this feature tries to repair the links.

Reply:
It looks for files on disk that are matched by the citation key (= "associated", probably a bad name), but which are not yet have this file in the `file` field (= "linked"). https://github.com/JabRef/jabref/blob/a13f3075122256535dba7f8c32ca4f196b68856c/src/main/java/org/jabref/gui/externalfiles/AutoSetFileLinksUtil.java#L88-L92

So it's not about existing links that no longer point to a valid file (I think we have an integrity check for this though).

Reply:
Oh ok. I totally misunderstood this. Sorry. I'll change that.